src/main/java/ToDoListFeature/ToDoList.java

```diff
@@ -33,83 +52,149 @@ private void unMarkTask(int index) {
         System.out.println("  " + this.toDoListArray.get(index -1).toString());
     }
 
+    /**
+     * Execute the ToDoList feature
+     * @param input the scanner to read user input
+     */
     public void execute(Scanner input) {
         featureIntroMessage();
         boolean isFinished = false;
         while (!isFinished) {
             String inputText = input.nextLine();
-            if (matchesPattern(inputText, MARK_FEATURE_PATTERN)) {
-                int numberPosition = inputText.indexOf(" ");
-                markTask(Integer.parseInt(inputText.substring(numberPosition + 1)));
-            } else if(matchesPattern(inputText, UNMARK_FEATURE_PATTERN)) {
-                int numberPosition = inputText.indexOf(" ");
-                unMarkTask(Integer.parseInt(inputText.substring(numberPosition + 1)));
-            } else {
-                switch (inputText) {
-                case "list":
-                    printToDoList();
-                    break;
-
-                case "^mark \\d+$":
-                    System.out.println("nice");
-                    break;
-
-                case "help":
-                    featureIntroMessage();
-                    break;
-
-                case "quit":
-                    return;
-
-                default:
-                    addItem(new ToDoItem(inputText));
-                    break;
+            try {
+                if (!processKeywordInput(inputText)) {
+                    //add a specific todoTask based on the keyword input
+                    switch (inputText) {
+                    case "list":
+                        printToDoList();
+                        break;
+                
+                    case "help":
+                        featureIntroMessage();
+                        break;
+                
+                    case "quit":
+                        return;
+                
+                    default:
+                        System.out.println("invalid command, use [help] to show all the commands");
+                        break;
+                    }
                 }
+                messageDivider();
+            } catch (StringIndexOutOfBoundsException e) {
+                System.out.println("invalid command, use [help] to show all the commands");
+                messageDivider();
             }
         }
     }
 
+    /**
+     * Print the ToDoList
+     */
     public void printToDoList() {
         System.out.println("Here are the tasks in your list:");
         System.out.println(toString());
-        messageDivider();
     }
 
-    public void addItem(ToDoItem item) {
+    /**
+     * Add a ToDo item to the ToDoList
+     * @param item the ToDo item to be added
+     */
+    public void addItem(ToDo item) {
```

Comment:
An alternative way to name the `ToDo item` would be `task`. It might enhance code readability